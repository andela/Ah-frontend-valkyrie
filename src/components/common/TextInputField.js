import React from 'react';
import PropTypes from 'prop-types';

const TextInputField = ( {
  name,
  label,
  type,
  placeholder,
  classname,
  icon,
  changed,
  required,
} ) => (
  <div className="form-group">
    <div className="input-group">
      <span className="input-group-addon">
        <i className={ icon } />
      </span>
      <input
        name={ name }
        label={ label }
        type={ type }
        placeholder={ placeholder }
        className={ classname }
        onChange={ changed }
        required={ required }
      />
    </div>
  </div>
);

TextInputField.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  icon: PropTypes.string,
  classname: PropTypes.string,
  changed: PropTypes.func,
  required: PropTypes.bool,
};
TextInputField.defaultProps = {
  label: '',
  placeholder: '',
  icon: '',
  classname: '',
  required: false,
  name: '',
  type: '',
  changed: null,
};

export default TextInputField;
