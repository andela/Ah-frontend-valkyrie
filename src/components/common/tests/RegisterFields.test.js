import React from "react";
import expect from "expect";
import { create } from "react-test-renderer";
import TextInputField from "../RegisterFields";

describe("TextInputField component", () => {
  test("it matches the snapshot", () => {
    const component = create(<TextInputField />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
