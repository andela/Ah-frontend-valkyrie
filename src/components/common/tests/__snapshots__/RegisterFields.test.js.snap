// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputField component it matches the snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group"
  >
    <span
      className="input-group-addon"
    >
      <i
        className=""
      />
    </span>
    <input
      className="form-control form-control-lg"
      icon=""
      label=""
      placeholder=""
    />
  </div>
</div>
`;
